TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/mediation/adapter/SmartRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package tv.teads.teadssdkdemo.format.mediation.adapter
+
+import android.content.Context
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.smartadserver.android.library.model.SASAdPlacement
+import com.smartadserver.android.library.ui.SASBannerView
+import com.smartadserver.android.library.util.SASConfiguration
+import tv.teads.sdk.TeadsMediationSettings
+import tv.teads.teadssdkdemo.component.GenericRecyclerViewAdapter
+import tv.teads.teadssdkdemo.data.RecyclerItemType
+
+/**
+ * Simple RecyclerView adapter
+ */
+class SmartRecyclerViewAdapter(context: Context?, title: String)
+    : GenericRecyclerViewAdapter(title) {
+    private val supplyChainObjectString: String = "" // "1.0,1!exchange1.com,1234,1,publisher,publisher.com";
+    private val siteID = 385317L
+    private val pageName = "1399205"
+    private val formatID = 96445L
+
+    private val smartAdView: SASBannerView
+
+    init {
+        smartAdView = SASBannerView(context!!)
```

Comment:
to void force unwrap, pass non nullable context to the adapter, then on the view instantiating it use requireContext or whatever. force wrappings make me nuts XD